LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -197,58 +271,113 @@ List<Module> GetModuleList()
 
         public void CreateClassrooms(int classroomCount)
         {
-            if(moduleIdList.Count == 0)
+            string action = "Create Classroom";
+            int i = 0;
+            ReportHeader(actionName: action);
+
+            if (ModuleIdList.Count == 0)
             {
-                ReportError("Modules", "Classroom");
+                ReportError(missingResource: "Modules", failedAction: action);
             }
             else
             {
-                ReportHeader("Create Classroom");
-                Guid moduleId = moduleIdList[AttributeGenerator.random.Next(0, moduleIdList.Count)];
+               
+                Guid moduleId = ModuleIdList[AttributeGenerator.random.Next(0, ModuleIdList.Count)];
```

Comment:
You are using the same module for all classrooms that you are creating. You need to choose different modules. Move these lines inside the while loop:
                Guid moduleId = ModuleIdList[AttributeGenerator.random.Next(0, ModuleIdList.Count)];
                Module module;
                moduleDictionary.TryGetValue(moduleId, out module);